presto-main/src/main/java/com/facebook/presto/sql/gen/JoinProbeCompiler.java

```diff
@@ -311,34 +330,16 @@ private static void generateGetChannelCountMethod(ClassDefinition classDefinitio
                 .retInt();
     }
 
-    private static void generateAppendToMethod(
-            ClassDefinition classDefinition,
-            CallSiteBinder callSiteBinder,
-            List<Type> types,
-            List<Integer> probeOutputChannels,
-            List<FieldDefinition> blockFields,
-            FieldDefinition positionField)
+    private static void generateGetOutputChannelsMethod(ClassDefinition classDefinition, FieldDefinition probeOutputChannelsField)
     {
-        Parameter pageBuilder = arg("pageBuilder", PageBuilder.class);
         MethodDefinition method = classDefinition.declareMethod(
                 a(PUBLIC),
-                "appendTo",
-                type(void.class),
-                pageBuilder);
-
+                "getOutputChannels",
+                type(int[].class));
         Variable thisVariable = method.getThis();
-        int pageBuilderOutputChannel = 0;
-        for (int outputChannel : probeOutputChannels) {
-            Type type = types.get(outputChannel);
-            method.getBody()
-                    .comment("%s.appendTo(block_%s, position, pageBuilder.getBlockBuilder(%s));", type.getClass(), outputChannel, pageBuilderOutputChannel)
-                    .append(constantType(callSiteBinder, type).invoke("appendTo", void.class,
```

Comment:
I guess `appendTo` was one of the primary reasons to have generated `JoinProbe` class. With your change, is the generated class still necessary? I'm aware it has the `probeHashChannel.isPresent()` check inlined, but

* does it matter? Does the generated class still outperform `SimpleJoinProbe`?
* maybe a simpler solution (two classes like `SimpleJoinProbe`, one expecting probe hash channel and one not) would yield same performance without generated code?